modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/portlet/action/UpdateOrganizationOrganizationSiteMVCActionCommand.java

```diff
@@ -0,0 +1,171 @@
+	},
+	service = MVCActionCommand.class
+)
+public class UpdateOrganizationOrganizationSiteMVCActionCommand
+	extends BaseMVCActionCommand {
+
+	@Override
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		try {
+			updateOrganizationSite(actionRequest);
+		}
+		catch (Exception e) {
+			String mvcPath = "/organization_site.jsp";
+
+			if (e instanceof NoSuchOrganizationException ||
+				e instanceof PrincipalException) {
+
+				SessionErrors.add(actionRequest, e.getClass());
+
+				mvcPath = "/error.jsp";
+			}
+
+			actionResponse.setRenderParameter("mvcPath", mvcPath);
+		}
+	}
+
+	protected void updateOrganizationSite(ActionRequest actionRequest)
+		throws Exception {
+
+			boolean site = ParamUtil.getBoolean(actionRequest, "site");
+
+			long publicLayoutSetPrototypeId = ParamUtil.getLong(
+				actionRequest, "publicLayoutSetPrototypeId");
+			long privateLayoutSetPrototypeId = ParamUtil.getLong(
+				actionRequest, "privateLayoutSetPrototypeId");
+			boolean publicLayoutSetPrototypeLinkEnabled = ParamUtil.getBoolean(
+				actionRequest, "publicLayoutSetPrototypeLinkEnabled",
+				publicLayoutSetPrototypeId > 0);
+			boolean privateLayoutSetPrototypeLinkEnabled = ParamUtil.getBoolean(
+				actionRequest, "privateLayoutSetPrototypeLinkEnabled",
+				privateLayoutSetPrototypeId > 0);
+
+			long organizationId = ParamUtil.getLong(
+				actionRequest, "organizationId");
+
+			Organization organization = _organizationService.getOrganization(
+				organizationId);
+
+			Group organizationGroup = organization.getGroup();
+
+			ThemeDisplay themeDisplay =
+				(ThemeDisplay)actionRequest.getAttribute(WebKeys.THEME_DISPLAY);
+
+			List<EmailAddress> emailAddresses =
+				_emailAddressService.getEmailAddresses(
+					Organization.class.getName(), organizationId);
+			List<Phone> phones = _phoneService.getPhones(
+				Organization.class.getName(), organizationId);
+			List<Website> websites = _websiteService.getWebsites(
+				Organization.class.getName(), organizationId);
+
+			boolean deleteLogo = false;
+
+			byte[] logoBytes = null;
+
+			if (organization.getLogoId() > 0) {
+				Image image = ImageLocalServiceUtil.getImage(
+					organization.getLogoId());
+
+				logoBytes = image.getTextObj();
+			}
+
+			organization = _organizationService.updateOrganization(
+				organizationId, organization.getParentOrganizationId(),
+				organization.getName(), organization.getType(),
+				organization.getRegionId(), organization.getCountryId(),
+				organization.getStatusId(), organization.getComments(),
+				deleteLogo, logoBytes, site, organization.getAddresses(),
+				emailAddresses,
+				OrgLaborServiceUtil.getOrgLabors(organizationId), phones,
+				websites, null);
+
+			if (GroupPermissionUtil.contains(
+					themeDisplay.getPermissionChecker(), organizationGroup,
+					ActionKeys.UPDATE)) {
+
+				_sites.updateLayoutSetPrototypesLinks(
```

Comment:
Your idea to move the relevant variable declarations inside this block is a good one.